src/api/vst3.rs

```diff
@@ -0,0 +1,757 @@
+                } else {
+                    0.0
+                }
+            }
+
+            unsafe fn get_param_normalized(&self, id: u32) -> f64 {
+                if let Some(param) = self.param(id) {
+                    self.plugin().get_parameter(param) as f64
+                } else {
+                    std::f64::NAN
+                }
+            }
+
+            unsafe fn set_param_normalized(&self, id: u32, value: f64) -> i32 {
+                if let Some(param) = self.param(id) {
+                    self.plugin().set_parameter(param, value as f32);
+                    0
+                } else {
+                    1
+                }
+            }
+
+            unsafe fn set_component_handler(&self, _handler: VstPtr<IComponentHandler>) -> i32 {
+                // we'll need to implement this once we have GUI integration, the _handler
+                // arg is used to begin/perform/complete edits.
+                0
+            }
+            unsafe fn create_view(&self, _name: FIDString) -> *mut std::os::raw::c_void {
+                std::ptr::null_mut()
+            }
+        }
+
+        impl IComponent for Vst3Wrapper {
+            unsafe fn get_controller_class_id(&self, tuid: *mut IID) -> i32 {
+                std::ptr::write(tuid, $iid);
+                0
+            }
+            
+            unsafe fn set_io_mode(&self, _mode: IoMode) -> i32 {
+                0
+            }
+
+            unsafe fn get_bus_count(&self, type_: MediaType, dir: BusDirection) -> i32 {
+                if type_ == 0 {
+                    // audio
+                    1 // single input/output
+                } else if type_ == 1 {
+                    if dir == 0 {
+                        // input
+                        1
+                    } else {
+                        // output, todo: midi output
+                        0
+                    }
+                } else {
+                    eprintln!("invalid argument passed to IComponent::get_bus_count");
+                    2
+                }
+            }
+
+            unsafe fn get_bus_info(
+                &self,
+                type_: MediaType,
+                dir: BusDirection,
+                index: i32,
+                out_info: *mut BusInfo,
+            ) -> i32 {
+                if index != 0 || out_info.is_null() || dir < 0 || dir > 1 || type_ < 0 || type_ > 1
+                {
+                    eprintln!("invallid argument passed to IComponent::get_bus_info");
+                    return 2; // kInvalidArgument
+                }
+                let is_input = dir == 0;
+                let is_event = type_ == 0;
+                let (name, channel_count) = match (is_input, is_event) {
+                    (true, false) => ("Audio Input", 2),
+                    (false, false) => ("Audio Output", 2),
+                    (true, true) => ("Event Input", 0),
```

Comment:
is it alright as a stopgap to use `Plugin::INPUT_CHANNELS`, `Plugin::OUTPUT_CHANNELS`, `Plugin::MIDI_INPUT` here?

until we figure out a more comprehensive bus configuration scheme, that is